client/pokemon/pokemon.ctrl.js

```diff
@@ -69,11 +71,12 @@ module.exports = class Pokemon {
       `${this.data.formName ? '-' + this.data.formName : ''}`;
   }
   parseAllProps () {
+    this.uploadGen = this.data.gen;
     this.parseBoxViewProps();
 
+    this.tidFull = this.data.tidFull;
```

Comment:
We'll need to remember this will be undefined for pre-1.3 uploads, but as long as we stick with our current implementation of only using tidFull for Gen 7 Pokémon, it's fine since this only assigns it a value of "undefined" on older stuff.